### PR TITLE
Create 3Sum.cs

### PR DESCRIPTION
**Pull request by rokovalenko**
I really don't know what to comment on here, so ask your questions.

### CODE REVIEW

**Comment by rokovalenko** [comment]
LeetCode stats
Runtime: 601 ms, faster than 17.14% of C# online submissions for 3Sum.
Memory Usage: 50.7 MB, less than 24.39% of C# online submissions for 3Sum.

**Comment by vlemish** [comment]
I have no comments, good solution 👍  It's strange that your 601 ms and mine 800 ms give almost the same percentage of runtime speed, it differs only by 2 percents.

File: rkovalenko/3Sum.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+public class Solution {
+    public IList<IList<int>> ThreeSum(int[] nums) {
+        Array.Sort(nums);
+        var hashSet = new HashSet<(int,int,int)>();
```
Comment (PShchegolevatykh) [inline rkovalenko/3Sum.cs:4]: Good idea to use hashset of tuples. I've tried using ordinary hashsets for triplets and duplicates were a problem like [0,0,0] shortened to just [0]. Tuples are good things to have here. :)

File: rkovalenko/3Sum.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        for(int i = 0; i < nums.Length - 1; ++i)
+        {
+            var j = i + 1;
+            var k = nums.Length - 1;
+
+            while (k < nums.Length)
+            {
+                if(k <= j || k <= i)
```
Comment (PShchegolevatykh) [inline rkovalenko/3Sum.cs:13]: nit: Maybe explain loop conditions and premature exit to make it obvious for readers next time. (I remember you asked what could have been commented)

File: rkovalenko/3Sum.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                }
+
+                var sum = nums[i] + nums[j] + nums[k];
+
+                if (sum == 0)
+                {
+                    hashSet.Add((nums[i], nums[j], nums[k]));
+                    --k;
+                }
+                if (sum < 0)
```
Comment (PShchegolevatykh) [inline rkovalenko/3Sum.cs:25]: Nice playing around properties of sorted sequences.

**Review by PShchegolevatykh — APPROVED** [review]

**Pull request merged**